test/lit.cfg

```diff
@@ -741,6 +742,62 @@ elif run_os == 'linux-gnu' or run_os == 'linux-gnueabihf' or run_os == 'freebsd'
     config.target_ld = (
         "ld -L%s" %
         (os.path.join(test_resource_dir, config.target_sdk_name)))
+elif run_os == 'linux-androideabi':
+    lit_config.note("Testing Android " + config.variant_triple)
+    config.target_object_format = "elf"
+    config.target_dylib_extension = "so"
+    config.target_runtime = "native"
+    config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
+    config.target_sdk_name = "android"
+    android_linker_opt = "-L {libcxx} -L {libgcc}".format(
+        libcxx=os.path.join(config.android_ndk_path,
+                            "sources", "cxx-stl", "llvm-libc++", "libs",
+                            "armeabi-v7a"),
+        libgcc=os.path.join(config.android_ndk_path,
+                            "toolchains",
+                            "arm-linux-androideabi-{}".format(
+                                config.android_ndk_gcc_version),
+                            "prebuilt", "linux-x86_64", "lib", "gcc",
+                            "arm-linux-androideabi",
+                            config.android_ndk_gcc_version))
+    config.target_build_swift = (
+        '%s -target %s -sdk %s %s -Xlinker -pie %s %s %s %s'
+        % (config.swiftc, config.variant_triple, config.variant_sdk,
+           android_linker_opt, resource_dir_opt, mcp_opt,
+           config.swift_test_options, swift_execution_tests_extra_flags))
+    config.target_swift_frontend = (
+        '%s -frontend -target %s -sdk %s %s %s'
+        % (config.swift, config.variant_triple, config.variant_sdk,
+           android_linker_opt, resource_dir_opt))
+    subst_target_swift_frontend_mock_sdk = config.target_swift_frontend
+    subst_target_swift_frontend_mock_sdk_after = ""
+    config.target_run = os.path.join(
+        config.swift_src_root, 'utils', 'android', 'adb_test_runner.py')
+    # FIXME: Include -sdk in this invocation.
+    config.target_sil_opt = (
+        '%s -target %s %s %s' %
+        (config.sil_opt, config.variant_triple, resource_dir_opt, mcp_opt))
+    config.target_swift_ide_test = (
+        '%s -target %s %s %s %s' %
+        (config.swift_ide_test, config.variant_triple, resource_dir_opt,
+         mcp_opt, ccp_opt))
+    subst_target_swift_ide_test_mock_sdk = config.target_swift_ide_test
+    subst_target_swift_ide_test_mock_sdk_after = ""
+    config.target_swiftc_driver = (
+        "%s -target %s -sdk %s %s %s %s" %
+        (config.swiftc, config.variant_triple, config.variant_sdk,
+         android_linker_opt, resource_dir_opt, mcp_opt))
+    config.target_swift_modulewrap = (
+        '%s -modulewrap -target %s' %
+        (config.swiftc, config.variant_triple))
+    config.target_clang = (
+        "clang++ -target %s %s" %
+        (config.variant_triple, clang_mcp_opt))
+    config.target_ld = (
+        "ld -L%s" %
```

Comment:
By toolchain, do you mean the Android NDK? That is, this linker should be `/path/to/android-ndk-r11c/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/arm-linux-androideabi/bin/ld.gold` (see note below)?